core-java-modules/core-java-compiler/src/test/java/com/baeldung/compilerApi/JavaCompilerTest.java

```diff
@@ -0,0 +1,154 @@
+package com.baeldung.compilerApi;
+
+import org.junit.jupiter.api.*;
+import org.junit.jupiter.api.io.TempDir;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class JavaCompilerTest {
+
+    @TempDir
+    static Path tempDir;
+
+    private JavaCompilerUtils compilerUtil;
+    private final ByteArrayOutputStream outputCaptor = new ByteArrayOutputStream();
+    private PrintStream standardOut;
+
+    @BeforeEach
+    void setUp() throws Exception {
+        // Create a specific output directory for compiled classes
+        Path outputDir = tempDir.resolve("classes");
+        Files.createDirectories(outputDir);
+
+        // Initialize the compiler util with the output directory
+        compilerUtil = new JavaCompilerUtils(outputDir);
+
+        // Set up System.out capture
+        standardOut = System.out;
+        System.setOut(new PrintStream(outputCaptor));
+    }
+
+    @AfterEach
+    void tearDown() {
+        // Restore System.out
+        System.setOut(standardOut);
+    }
```

Comment:
Don't do this manually, use something like https://github.com/stefanbirkner/system-lambda or similar